src/wallet/wallet.h

```diff
@@ -139,6 +158,20 @@ class CKeyPool
             READWRITE(nVersion);
         READWRITE(nTime);
         READWRITE(vchPubKey);
+        if (ser_action.ForRead()) {
+            try {
+                READWRITE(FLATDATA(type));
+                READWRITE(m_pre_split);
+            } catch (std::ios_base::failure&) {
+                /* Set as external address if we can't read the type boolean
+                   (this will be the case for any wallet before the HD chain split version) */
+                type = HDChain::ChangeType::EXTERNAL;
+                m_pre_split = false;
```

Comment:
shouldn't this be `true` ?
Also we can just remove it all together if we are not going to have HD keys pre-HD_SPLIT.

Reply:
yeah, done.